core/src/main/java/apoc/export/arrow/ExportArrowUtil.java

```diff
@@ -0,0 +1,228 @@
+package apoc.export.arrow;
+
+import apoc.export.util.ProgressReporter;
+import apoc.meta.Meta;
+import apoc.util.JsonUtil;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.UInt8Vector;
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.dictionary.Dictionary;
+import org.apache.arrow.vector.dictionary.DictionaryEncoder;
+import org.apache.arrow.vector.dictionary.DictionaryProvider;
+import org.apache.arrow.vector.ipc.ArrowStreamWriter;
+import org.apache.arrow.vector.ipc.ArrowWriter;
+import org.apache.arrow.vector.types.pojo.DictionaryEncoding;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.neo4j.cypher.export.SubGraph;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Path;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.Result;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.nio.channels.Channels;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import static apoc.export.arrow.ArrowConstants.END_FIELD;
+import static apoc.export.arrow.ArrowConstants.ID_FIELD;
+import static apoc.export.arrow.ArrowConstants.LABELS_FIELD;
+import static apoc.export.arrow.ArrowConstants.START_FIELD;
+import static apoc.export.arrow.ArrowConstants.STREAM_EDGE_PREFIX;
+import static apoc.export.arrow.ArrowConstants.STREAM_NODE_PREFIX;
+import static apoc.export.arrow.ArrowConstants.TYPE_FIELD;
+import static apoc.util.Util.labelString;
+
+public class ExportArrowUtil {
+
+    protected static void implementArrowExport(RootAllocator allocator, int batchSize, Object valueToProcess, ArrowConstants.FunctionType function, ProgressReporter reporter, OutputStream fd) {
+        DictionaryProvider.MapDictionaryProvider dictProvider = new DictionaryProvider.MapDictionaryProvider();
+        AtomicInteger index = new AtomicInteger();
+        Map<String, FieldVector> vectorMap = new TreeMap<>();
+
+        switch (function) {
+            case RESULT:
+                String[] header = ((Result) valueToProcess).columns().toArray(new String[((Result) valueToProcess).columns().size()]);
+                ((Result) valueToProcess).accept((row) -> {
+                    for (String keyName : header) {
+                        Object value = row.get(keyName);
+                        writeArrowResult(reporter, allocator, vectorMap, index, value, dictProvider, batchSize, fd);
+                    }
+                    return true;
+                });
```

Comment:
Riguardalo il codice, è brutto scritto come sopra, possiamo mettere a fattor comune così:
```suggestion
                Result result = ((Result) valueToProcess);
                final List<String> columns = ((Result) valueToProcess).columns();
                String[] header = columns.toArray(new String[columns.size()]);
                result.accept((row) -> {
                    for (String keyName : header) {
                        Object value = row.get(keyName);
                        writeArrowResult(reporter, allocator, vectorMap, index, value, dictProvider, batchSize, fd);
                    }
                    return true;
                });
```